diff --git a/node_modules/@capacitor/push-notifications/ios/Sources/PushNotificationsPlugin/PushNotificationsHandler.swift b/node_modules/@capacitor/push-notifications/ios/Sources/PushNotificationsPlugin/PushNotificationsHandler.swift
index f3972b6..c3bb892 100644
--- a/node_modules/@capacitor/push-notifications/ios/Sources/PushNotificationsPlugin/PushNotificationsHandler.swift
+++ b/node_modules/@capacitor/push-notifications/ios/Sources/PushNotificationsPlugin/PushNotificationsHandler.swift
@@ -29,7 +29,7 @@ public class PushNotificationsHandler: NSObject, NotificationHandlerProtocol {
             }
         }
 
-        if let optionsArray = self.plugin?.getConfig().getArray("presentationOptions") as? [String] {
+        if let optionsArray = self.plugin?.getConfig().getConfigJSON()["presentationOptions"] as? [String] {
             var presentationOptions = UNNotificationPresentationOptions.init()
 
             optionsArray.forEach { option in
@@ -83,7 +83,7 @@ public class PushNotificationsHandler: NSObject, NotificationHandlerProtocol {
             "subtitle": request.content.subtitle,
             "badge": request.content.badge ?? 1,
             "body": request.content.body,
-            "data": JSTypes.coerceDictionaryToJSObject(request.content.userInfo) ?? [:]
+            "data": JSTypes.coerceDictionaryToJSObject(request.content.userInfo as? [String: Any]) ?? [:]
         ]
     }
 }
diff --git a/node_modules/@capacitor/push-notifications/ios/Sources/PushNotificationsPlugin/PushNotificationsPlugin.swift b/node_modules/@capacitor/push-notifications/ios/Sources/PushNotificationsPlugin/PushNotificationsPlugin.swift
index 3b7c268..462725e 100644
--- a/node_modules/@capacitor/push-notifications/ios/Sources/PushNotificationsPlugin/PushNotificationsPlugin.swift
+++ b/node_modules/@capacitor/push-notifications/ios/Sources/PushNotificationsPlugin/PushNotificationsPlugin.swift
@@ -78,11 +78,11 @@ public class PushNotificationsPlugin: CAPPlugin, CAPBridgedPlugin {
         self.notificationDelegateHandler.requestPermissions { granted, error in
             guard error == nil else {
                 if let err = error {
-                    call.reject(err.localizedDescription)
+                    call.reject(err.localizedDescription, nil, nil, [:])
                     return
                 }

-                call.reject("unknown error in permissions request")
+                call.reject("unknown error in permissions request", nil, nil, [:])
                 return
             }

@@ -123,7 +123,7 @@ public class PushNotificationsPlugin: CAPPlugin, CAPBridgedPlugin {
      */
     @objc func getDeliveredNotifications(_ call: CAPPluginCall) {
         if !appDelegateRegistrationCalled {
-            call.reject("event capacitorDidRegisterForRemoteNotifications not called.  Visit https://capacitorjs.com/docs/apis/push-notifications for more information")
+            call.reject("event capacitorDidRegisterForRemoteNotifications not called.  Visit https://capacitorjs.com/docs/apis/push-notifications for more information", nil, nil, [:])
             return
         }
         UNUserNotificationCenter.current().getDeliveredNotifications(completionHandler: { (notifications) in
@@ -141,11 +141,11 @@ public class PushNotificationsPlugin: CAPPlugin, CAPBridgedPlugin {
      */
     @objc func removeDeliveredNotifications(_ call: CAPPluginCall) {
         if !appDelegateRegistrationCalled {
-            call.reject("event capacitorDidRegisterForRemoteNotifications not called.  Visit https://capacitorjs.com/docs/apis/push-notifications for more information")
+            call.reject("event capacitorDidRegisterForRemoteNotifications not called.  Visit https://capacitorjs.com/docs/apis/push-notifications for more information", nil, nil, [:])
             return
         }
-        guard let notifications = call.getArray("notifications", JSObject.self) else {
-            call.reject("Must supply notifications to remove")
+        guard let notifications = call.getArray("notifications") as? [JSObject] else {
+            call.reject("Must supply notifications to remove", nil, nil, [:])
             return
         }

@@ -159,7 +159,7 @@ public class PushNotificationsPlugin: CAPPlugin, CAPBridgedPlugin {
      */
     @objc func removeAllDeliveredNotifications(_ call: CAPPluginCall) {
         if !appDelegateRegistrationCalled {
-            call.reject("event capacitorDidRegisterForRemoteNotifications not called.  Visit https://capacitorjs.com/docs/apis/push-notifications for more information")
+            call.reject("event capacitorDidRegisterForRemoteNotifications not called.  Visit https://capacitorjs.com/docs/apis/push-notifications for more information", nil, nil, [:])
             return
         }
         UNUserNotificationCenter.current().removeAllDeliveredNotifications()
